src/ZenodoFile.ts

```diff
@@ -2,36 +2,27 @@ import { z } from 'zod';
 
 import type { Zenodo } from './Zenodo';
 
-export class ZenodoFile {
-  private zenodo: Zenodo;
-  id: string;
-  filename: string;
-  filesize: number;
-  checksum: string; // md5
-  links: {
-    download: string;
-    self: string;
-  };
+// Define the Zod schema
+const zenodoFileSchema = z.object({
+  id: z.string(),
+  filename: z.string(),
+  filesize: z.number(),
+  checksum: z.string(),
+  links: z.object({
+    download: z.string().url(),
+    self: z.string().url(),
+  }),
+});
+
+type ZenodoFileType = z.infer<typeof zenodoFileSchema>;
 
-  static schema = z.object({
-    id: z.string(),
-    filename: z.string(),
-    filesize: z.number(),
-    checksum: z.string().regex(/^[a-fA-F0-9]{32}$/, 'Invalid MD5 checksum'),
-    links: z.object({
-      download: z.string().url(),
-      self: z.string().url(),
-    }),
-  });
+export class ZenodoFile implements ZenodoFileType {
+  private zenodo: Zenodo;
 
   constructor(zenodo: Zenodo, file: unknown) {
-    const validatedFile = ZenodoFile.schema.parse(file);
+    const validatedFile = zenodoFileSchema.parse(file);
     this.zenodo = zenodo;
-    this.id = validatedFile.id;
-    this.filename = validatedFile.filename;
-    this.filesize = validatedFile.filesize;
-    this.checksum = validatedFile.checksum;
-    this.links = validatedFile.links;
+    Object.assign(this, validatedFile);
```

Comment:
I try to use Zod to validate the data coming from the API. The tests are passing but I think I lose types with Object.assign.
I get a TS error
![image](https://github.com/user-attachments/assets/49f75e74-762d-40b3-87ef-0a39f13a6737)

Reply:
Not really related to the migration, but you need to declare the properties:

```ts
public id: string;
// ...
```

`implements` doesn't do it for you.